core/vm/intpool.go

```diff
@@ -65,3 +71,54 @@ func (p *intPool) put(is ...*big.Int) {
 		p.pool.push(i)
 	}
 }
+
+const poolPoolLimit = 5
+const (
+	PoolPoolTypeMiner = iota
+	PoolPoolTypeImporter
+	PoolPoolTypeTracer
+	PoolPoolTypeRPC
+	PoolPoolTypes
+)
+
+// IntPoolPool manages a pool of intPools
+type IntPoolPool struct {
+	pools [PoolPoolTypes][poolPoolLimit]*intPool
+
+	// Mutex
+	locks [PoolPoolTypes]sync.RWMutex
+}
+
+var intPoolPool = IntPoolPool{}
```

Comment:
@karalabe I would like to export this but I'm looking for a nicer name. Any idea?

Reply:
don't export it. not even the type. It's a very specialized thing used by the vm, no need to surface it.